File: AtCoder/ABC/C/268/C.rb
n = gets.chomp.to_i
hash = Hash.new(0)
gets.chomp.split.map(&:to_i).each_with_index do |p, i|
    key = i - p
    3.times do
        key += 1
        key %= n
        hash[key] += 1
    end
end
ans = 0
hash.each do |key, val|
    ans = [ans, val].max
end
p ans

File: AtCoder/ABC/C/268/C.py
n = int(input())
p = list(map(int, input().split()))
d = dict()
for i in range(n):
    key = i - p[i]
    for j in range(3):
        key += 1
        key %= n
        if not key in d: d[key] = 0
        d[key] += 1

ans = 0
for key, val in d.items():
    ans = max(ans, val)
print(ans)